sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -342,6 +342,16 @@ namespace Azure { namespace Core { namespace Http {
      */
     bool m_keepAlive = true;
 
+    /**
+     * @brief Implement #BodyStream `OnRead`. Calling this function pulls data from the wire.
+     *
+     * @param context #Context so that operation can be cancelled.
+     * @param buffer Buffer where data from wire is written to.
+     * @param count The number of bytes to read from the network.
+     * @return The actual number of bytes read from the network.
+     */
+    int64_t OnRead(Azure::Core::Context const& context, uint8_t* buffer, int64_t count) override;
```

Comment:
This should be private.

Reply:
It makes sense to be explicit and add the `private:` keyword, but just to confirm, isn't this already private given defaults in C++ classes is private, and this is written before the `public:` keyword?

Reply:
Yes, it is already private.
The class declaration before it makes it a little confusing. But this is a private member next to a private class definition

Reply:
> The class declaration before it makes it a little confusing. But this is a private member next to a private class definition

Maybe we can move things around a bit to make it clearer. Can we move the nested class at the beginning or end, rather than declaring a class in the middle of a list of public members for the outer class?